cmd/update-main.go

```diff
@@ -46,238 +55,201 @@ USAGE:
 FLAGS:
   {{range .Flags}}{{.}}
   {{end}}
-EXAMPLES:
-   1. Check for any new official release.
-      $ minio {{.Name}}
-`,
+VERSION:
+  ` + Version + `{{"\n"}}`,
 }
 
-// update URL endpoints.
-const (
-	minioUpdateStableURL = "https://dl.minio.io/server/minio/release"
-)
-
-// updateMessage container to hold update messages.
-type updateMessage struct {
-	Update    bool          `json:"update"`
-	Download  string        `json:"downloadURL"`
-	NewerThan time.Duration `json:"newerThan"`
-}
+const releaseTagTimeLayout = "2006-01-02T15-04-05Z"
 
-// String colorized update message.
-func (u updateMessage) String() string {
-	if !u.Update {
-		updateMessage := color.New(color.FgGreen, color.Bold).SprintfFunc()
-		return updateMessage("You are already running the most recent version of ‘minio’.")
-	}
-	msg := colorizeUpdateMessage(u.Download, u.NewerThan)
-	return msg
-}
+const minioReleaseURL = "https://dl.minio.io/server/minio/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/"
 
-func parseReleaseData(data string) (time.Time, error) {
-	releaseStr := strings.Fields(data)
-	if len(releaseStr) < 2 {
-		return time.Time{}, errors.New("Update data malformed")
-	}
-	releaseDate := releaseStr[1]
-	releaseDateSplits := strings.SplitN(releaseDate, ".", 3)
-	if len(releaseDateSplits) < 3 {
-		return time.Time{}, (errors.New("Update data malformed"))
-	}
-	if releaseDateSplits[0] != globalMinioDefaultOwnerID {
-		return time.Time{}, (errors.New("Update data malformed, missing minio tag"))
-	}
-	// "OFFICIAL" tag is still kept for backward compatibility.
-	// We should remove this for the next release.
-	if releaseDateSplits[1] != "RELEASE" && releaseDateSplits[1] != "OFFICIAL" {
-		return time.Time{}, (errors.New("Update data malformed, missing RELEASE tag"))
+func _getCurrentReleaseTime(minioVersion, minioBinaryPath string) (releaseTime time.Time, err error) {
```

Comment:
Closure can't help to make a function unit testable. I will rename `_getCurrentReleaseTime()` into `getCurrentReleaseTime()` and `getCurrentReleaseTime()` into `GetCurrentReleasetime()`